### PR TITLE
Removing multiple reconciliation happening in the CC tests 

### PR DESCRIPTION
**Pull request by ShubhamRwt**
### Type of change

- Enhancement 

### Description

This PR fixes the issue mentioned in the issue #6799. It removes running the entire state machine to reach a state and then apply the wrong annotation. The changes now sets a new status in the KafkaRebalance resource with the desired state.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by ShubhamRwt** [comment]
Fixing the problem hence changed to draft

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java [hunk 1]
```diff
@@ -138,6 +139,8 @@ private KafkaRebalance createKafkaRebalance(KafkaRebalanceState currentState,
      * @param kcRebalance The Kafka Rebalance instance that will be returned by the resourceSupplier.
      * @return A future for the {@link KafkaRebalanceStatus} returned by the {@link KafkaRebalanceAssemblyOperator#computeNextStatus} method
      */
+
+
```
Comment (kyguy) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java:143]: Extra lines

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java [hunk 2]
```diff
@@ -175,6 +178,79 @@ public String cruiseControlHost(String clusterName, String clusterNamespace) {
...
+        when(mockCmOps.getAsync(CLUSTER_NAMESPACE, RESOURCE_NAME)).thenReturn(Future.succeededFuture(new ConfigMap()));
+        when(mockRebalanceOps.get(CLUSTER_NAMESPACE, RESOURCE_NAME)).thenReturn(kcRebalance);
+        when(mockRebalanceOps.getAsync(CLUSTER_NAMESPACE, RESOURCE_NAME)).thenReturn(Future.succeededFuture(kcRebalance));
+
+        return kcrao.computeNextStatus(recon, HOST, client, kcRebalance, currentState, initialAnnotation, rbOptions)
+                .compose(result -> {
+                    context.verify(() -> {
+                        assertThat(result.getStatus().getConditions(), StateMatchers.hasStateInConditions(nextState));
+                        System.out.println(result.getStatus());
```
Comment (kyguy) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java:247]: Is this a leftover from debugging?

Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java:247]: Yeah. Thanks for catching this

Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java]: Fixed @kyguy . Also it looks like I didn't removed the test from `KRAO` test class. I have now updated the PR

**Review by kyguy — APPROVED** [review]
Looks great @ShubhamRwt! Have a couple minor nits, feel free to ignore if they are the way they are supposed to be!

**Review by scholzj — APPROVED** [review]

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java [hunk 3]
```diff
@@ -175,6 +177,92 @@ public String cruiseControlHost(String clusterName, String clusterNamespace) {
                 });
     }
 
+    @Test
```
Comment (tomncooper) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java:180]: Just a nit but these tests should be placed at the bottom with the other tests below the utility methods.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java [hunk 4]
```diff
@@ -175,6 +177,92 @@ public String cruiseControlHost(String clusterName, String clusterNamespace) {
...
+     * @param vertx The vertx test instance.
+     * @param context The test context instance.
+     * @param currentState The current state of the resource before being passed to computeNextStatus.
+     * @param nextState The expected state of the resouce after computeNextStatus has been called.
+     * @param initialAnnotation The initial annotation attached to the Kafka Rebalance resource. For example none or refresh.
+     * @param kcRebalance The Kafka Rebalance instance that will be returned by the resourceSupplier.
+     * @return A future for the {@link KafkaRebalanceStatus} returned by the {@link KafkaRebalanceAssemblyOperator#computeNextStatus} method
+     */
+
+    private Future<KafkaRebalanceStatus> checkTransitionWithWrongAnnotation(Vertx vertx, VertxTestContext context,
```
Comment (tomncooper) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java:228]: Almost all of this method is identitcal to the other checkTransition method? Can you either alter that method to take an annotation option or extract the common code to another method?

**Review by tomncooper — CHANGES_REQUESTED** [review]
I think there needs to be a bit of clean up in the StateMachine test class.

**Review by tomncooper — APPROVED** [review]
LGTM

**Pull request merged**